qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -207,10 +209,32 @@ def jobs(self, limit=50, skip=0, status=None, db_filter=None):
         if db_filter:
             # status takes precedence over db_filter for same keys
             api_filter = {**db_filter, **api_filter}
-        job_info_list = self._api.get_status_jobs(limit=limit, skip=skip,
-                                                  filter=api_filter)
+
+        # Retrieve the requested number of jobs, using pagination. The API
+        # might limit the number of jobs per request.
+        job_responses = []
+        current_page_limit = limit
+
+        while True:
+            job_page = self._api.get_status_jobs(limit=current_page_limit,
+                                                 skip=skip, filter=api_filter)
+            job_responses += job_page
+            skip = skip + len(job_page)
+
+            if not job_page:
+                # Stop if there are no more jobs returned by the API.
+                break
+
+            if limit:
+                if len(job_responses) >= limit:
+                    # Stop if we have reached the limit.
+                    break
+                current_page_limit = limit - len(job_responses)
+            else:
```

Comment:
Does `limit=0` (or `None`) have special meanings (e.g. unlimited)? If it just means don't retrieve any jobs (which would make this method call useless), then shouldn't we just check up front and return without making any api calls? If it has special meanings it should be documented.